Add element resolver for keeping a live reference to an element in the same DOM tree (#2732)
adobe/spectrum-web-components: tools/reactive-controllers/element-resolution.md

## Description

> An `ElementResolutionController` keeps an active reference to another element in the same DOM tree. Supply the controller with the a selector by which it can locate the element to reference and it will manage observing the DOM tree in question to ensure that the reference it holds is always the first element that matches that selector or `null`.

Reviewer: ```suggestion
An `ElementResolutionController` keeps an active reference to another element in the same DOM tree. Supply the controller with a selector that matches the element and it will manage observing the DOM tree to ensure that the reference it holds is always the first matched element or `null`.
```

```

## Example

An `ElementResolutionController` can be applied to a host element like the following.

```js
import { html, LitElement } from 'lit';
import { ElementResolutionController } from '@spectrum-web-components/reactive-controllers/ElementResolution.js';

class RootEl extends LitElement {
    resolvedElement = new ElementResolutionController(this);

    costructor() {
        super();
        this.resovledElement.selector = '.other-element';
    }
}

customElements.define('root-el', RootEl);
```

In this example, the selector `'.other-element'` is supplied to the resolver, which mean in the following example, `this.resolvedElement.element` will maintain a reference to the sibling `<div>` element:

```html-no-demo
<root-el></root-el>
<div class="other-element"></div>
```

The resolved reference will always be the first element matching the selector applied, so in the following example the element with content "First!" will be the reference:

```html-no-demo
<root-el></root-el>
<div class="other-element">First!</div>
<div class="other-element">Last.</div>
```

> A [`MutationObserver`](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver) it leveraged to track mutations to the DOM tree in which the host element resides in order to update the element reference on any changes to the content therin that could change the resolved element.

Reviewer: typo: A MutationObserver _is_

Author: ```suggestion
A [`MutationObserver`](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver) is leveraged to track mutations to the DOM tree in which the host element resides in order to update the element reference on any changes to the content therein that could change the resolved element.
```
